gnovm/pkg/packages/resolve.go

```diff
@@ -0,0 +1,481 @@
+	}
+	modDir, err := findModDir(meta.Root)
+	if os.IsNotExist(err) {
+		return nil, errors.New("failed to clone mod")
+	} else if err != nil {
+		return nil, fmt.Errorf("failed to find parent module: %w", err)
+	}
+	modFilePath := filepath.Join(modDir, ModfileName)
+	modFileBytes, err := os.ReadFile(modFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read modfile: %w", err)
+	}
+	modFile, err := modfile.ParseLax(modFilePath, modFileBytes, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse modfile: %w", err)
+	}
+	meta.Module = &Module{
+		Path:   modFile.Module.Mod.Path,
+		Dir:    modDir,
+		GnoMod: modFilePath,
+	}
+
+	pkg, err := fillPackage(&meta)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fill Package %q: %w", meta.PkgPath, err)
+	}
+
+	return pkg, nil
+}
+
+func fillPackage(meta *PackageSummary) (*Package, error) {
+	fsFiles := []string{}
+	files := []string{}
+	fsTestFiles := []string{}
+	testFiles := []string{}
+	fsFiletestFiles := []string{}
+	filetestFiles := []string{}
+
+	pkgDir := meta.Root
+	pkgPath := meta.PkgPath
+
+	dir, err := os.ReadDir(pkgDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read module files list: %w", err)
+	}
+	for _, entry := range dir {
+		if entry.IsDir() {
+			continue
+		}
+
+		fileName := entry.Name()
+		if IsGnoTestFile(fileName) {
+			fsTestFiles = append(fsTestFiles, filepath.Join(pkgDir, fileName))
+			testFiles = append(testFiles, fileName)
+		} else if IsGnoFiletestFile(fileName) {
+			fsFiletestFiles = append(fsFiletestFiles, filepath.Join(pkgDir, fileName))
+			filetestFiles = append(filetestFiles, fileName)
+		} else if IsGnoFile(fileName) {
+			fsFiles = append(fsFiles, filepath.Join(pkgDir, fileName))
+			files = append(files, fileName)
+		}
+	}
+	name, imports, err := resolveNameAndImports(fsFiles)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve name and imports for %q: %w", pkgPath, err)
+	}
+	_, testImports, err := resolveNameAndImports(fsTestFiles)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve test name and imports for %q: %w", pkgPath, err)
+	}
+	_, filetestImports, err := resolveNameAndImports(fsFiletestFiles)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve filetest name and imports for %q: %w", pkgPath, err)
+	}
+
+	module := Module{}
+	if meta.Module != nil {
+		module = *meta.Module
+	}
+
+	return &Package{
+		ImportPath:       pkgPath,
+		Dir:              pkgDir,
+		Name:             name,
+		Module:           module,
+		Match:            meta.Match,
+		GnoFiles:         files,
+		Imports:          imports,
+		TestGnoFiles:     testFiles,
+		TestImports:      testImports,
+		FiletestGnoFiles: filetestFiles,
+		FiletestImports:  filetestImports,
+	}, nil
+}
+
+type Package struct {
+	Dir              string   `json:",omitempty"`
+	ImportPath       string   `json:",omitempty"`
+	Name             string   `json:",omitempty"`
+	Root             string   `json:",omitempty"`
```

Comment:
add comments to explain the fields.

this one for instance, is it the "location of the package that can be either specified locally or automatically cached on the disk"? or something else?

Reply:
done in [457b0c9](https://github.com/gnolang/gno/pull/2932/commits/457b0c9f4104ec67bb4beebd1ef7e503523b603c)